src/IK/src/InverseKinematics.cpp

```diff
@@ -0,0 +1,178 @@
+#include <BiomechanicalAnalysis/IK/InverseKinematics.h>
+#include <BiomechanicalAnalysis/Logging/Logger.h>
+#include <BipedalLocomotion/Conversions/ManifConversions.h>
+#include <iostream>
+
+using namespace BiomechanicalAnalysis::IK;
+using namespace BipedalLocomotion::ContinuousDynamicalSystem;
+using namespace BipedalLocomotion::Conversions;
+using namespace std::chrono_literals;
+
+bool HumanIK::initialize(std::weak_ptr<const BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
+                std::shared_ptr<iDynTree::KinDynComputations> kinDyn)
+{
+    constexpr std::size_t highPriority = 0;
+    constexpr std::size_t lowPriority = 1;
+    constexpr auto logPrefix = "[HumanIK::initialize]";
+
+    if ((kinDyn == nullptr) || (!kinDyn->isValid()))
+    {
+        BiomechanicalAnalysis::log()->error("{} Invalid kinDyn object.", logPrefix);
+        return false;
+    }
+
+    m_kinDyn = kinDyn;
+
+    m_jointPositions.resize(m_kinDyn->getNrOfDegreesOfFreedom());
+    m_jointVelocities.resize(m_kinDyn->getNrOfDegreesOfFreedom());
+
+    kinDyn->getRobotState(m_basePose, m_jointPositions, m_baseVelocity, m_jointVelocities, m_gravity);
+
+    m_system.dynamics = std::make_shared<FloatingBaseSystemKinematics>();
+    m_system.dynamics->setState({m_basePose.topRightCorner<3, 1>(), toManifRot(m_basePose.topLeftCorner<3,3>()) , m_jointPositions});
+
+    m_system.integrator = std::make_shared<ForwardEuler<FloatingBaseSystemKinematics>>();
+    m_system.integrator->setDynamicalSystem(m_system.dynamics);
+
+    auto ptr = handler.lock();
+    if (ptr == nullptr)
+    {
+        BiomechanicalAnalysis::log()->error("{} Invalid parameters handler.", logPrefix);
+        return false;
+    }
+
+    bool ok = m_qpIK.initialize(ptr->getGroup("IK"));
+    auto group = ptr->getGroup("IK").lock();
+    std::string variable;
+    group->getParameter("robot_velocity_variable_name", variable);
+    m_variableHandler.addVariable(variable, kinDyn->getNrOfDegreesOfFreedom() + 6);
+
+    m_link1OrientationTask = std::make_shared<BipedalLocomotion::IK::SO3Task>();
+    ok = ok && m_link1OrientationTask->setKinDyn(kinDyn);
+    ok = ok && m_link1OrientationTask->initialize(ptr->getGroup("LINK1_TASK"));
+    ok = ok && m_qpIK.addTask(m_link1OrientationTask, "link1_task", highPriority);
+
+    m_qpIK.finalize(m_variableHandler);
+
+    return ok;
+}
+
+bool HumanIK::setDt(const double dt)
+{
+    dt;
+    m_dtIntegration = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(dt));
+
+    return m_system.integrator->setIntegrationStep(m_dtIntegration);
+}
+
+double HumanIK::getDt() const
+{
+    return m_dtIntegration.count();
+}
```

Comment:
This will be returned in nanoseconds. Is this wanted?

Reply:
No, probably it is better to return the dt in seconds.

Reply:
I changed it and now it returns the value in seconds.